infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -506,7 +510,6 @@ private void onLeadershipAcquire() {
                 log.info("Start as Sink (receiver)");
                 interClusterReplicationService.getLogReplicationServer().getSinkManager().reset();
                 interClusterReplicationService.getLogReplicationServer().setLeadership(true);
-                statusFlag = false;
```

Comment:
we cannot remove setting the flag here because of the following case of role change:

Active(init status, flag = true) -> standby(flag = true) -> Active(flag = true, which is incorrect.  Status must be initialized again)